### PR TITLE
Added mpi backend plugin message

### PR DESCRIPTION
**Pull request by yindia**
# TL;DR
- Added message for new backend plugin MPI 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: protos/flyteidl/plugins/mpi.proto [hunk 1]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package flyteidl.plugins;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
+
+// Custom proto for plugin that enables distributed training using https://github.com/kubeflow/mpi-operator
+message DistributedMPITrainingTask {
+  // number of worker,launcher replicas spawned in the cluster for this job
```
Comment (kumare3) [inline protos/flyteidl/plugins/mpi.proto:9]: you need to add a comment to each field, that will automatically generate in the docs

File: protos/flyteidl/plugins/mpi.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// Custom proto for plugin that enables distributed training using https://github.com/kubeflow/mpi-operator
+message DistributedMPITrainingTask {
+  // number of worker spawned in the cluster for this job
+  int32 workers = 1;
+
+  // number of launcher replicas spawned in the cluster for this job
+  int32 launcher_replicas = 2;
+
+  // number of slots per worker used in hostfile.
+  int32 slots = 3;
```
Comment (EngHabu) [inline protos/flyteidl/plugins/mpi.proto:16]:
```suggestion
  int32 slots_per_worker = 3;
```

or
```suggestion
  int32 worker_slots = 3;
```

File: protos/flyteidl/plugins/mpi.proto [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
+
+// Custom proto for plugin that enables distributed training using https://github.com/kubeflow/mpi-operator
+message DistributedMPITrainingTask {
+  // number of worker spawned in the cluster for this job
+  int32 workers = 1;
+
+  // number of launcher replicas spawned in the cluster for this job
+  int32 launcher_replicas = 2;
```
Comment (EngHabu) [inline protos/flyteidl/plugins/mpi.proto:13]: Is there a relationship between number of workers and launcher_replicas?

Naming-wise, should they be `workers` and `launchers` or `worker_replicas` and `launcher_replicas` ?

Comment (evalsocket) [reply protos/flyteidl/plugins/mpi.proto:13]: no in operator both are replica. They are part of a common spec but both can have different value  https://github.com/kubeflow/mpi-operator/blob/a566d1d18046fe7f756b0ec7004693c898d589a0/pkg/apis/kubeflow/v1/types.go#L53
